command/connect/envoy/exec_test.go

```diff
@@ -264,3 +266,24 @@ func limitProcessLifetime(dur time.Duration) {
 		os.Exit(99)
 	})
 }
+
+func TestMakeBootstrapPipe_DoesNotBlockOnAFullPipe(t *testing.T) {
+	// A named pipe can buffer up to 64k, use a value larger than that
+	bootstrap := bytes.Repeat([]byte("a"), 66000)
+
+	execArgs = func(args ...string) (string, []string, error) {
+		args = append([]string{"run", "../../.."}, args...)
+		return "go", args, nil
+	}
```

Comment:
I know we have that other "self-exec" pattern for testing things like this, but despite having read about and seen that pattern a few times now, it still seems very confusing. There is tons of setup code requires, and it's not at all obvious how it all works.

I think this is a much better pattern for doing the same thing. It should be clear that this executes the current code using `go run`.

Reply:
> It should be clear that this executes the code in the git checkout using go run.

I've not used `go run` like this before so I actually found it a little hard to reason about what happens here.

I think I get it though:
 1. `../../..` ends up resolving to the repo root (is that reliable depending on where you run the test from?)
 2. `go run <dir>` will effectively build Consul's main.go in the root and so build the entire Consul binary and execute it (i've only ever used `go run` with individual named main.go files before not whole projects).

Is that right? I do think that has a big advantage of not relying on the right Consul binary being in the path. 

Perhaps a comment or two would help with the understandability goal for other readers who've not used `go run` in this way before or understand the guarantees of relative directory references during test runs?

Reply:
> is that reliable depending on where you run the test from?

Yes, `go test` always changes to the test directory. The only way it can break is if you build the test binary and run it from some other path, but that is rare, and I've seen people recommend that the binary run with a PWD of the test directory anyway to preserve that expectation.

> Is that right?

That is correct.

I can add a comment, however both of these things are pretty standard behaviour of the go toolchain.